examples/neural_compressor/language-modeling/run_clm.py

```diff
@@ -724,24 +716,31 @@ def compute_metrics(eval_preds):
     if optim_args.apply_quantization and optim_args.quantization_approach in {"static", "dynamic", "weight_only"}:
         model = trainer.model if isinstance(trainer.model, PreTrainedModel) else trainer.model._model
         quantizer = INCQuantizer.from_pretrained(model)
-        if optim_args.quantization_approach in ["static", "weight_only"]:
+        if optim_args.quantization_approach == "static":
             num_calibration_samples = min(len(train_dataset), optim_args.num_calibration_samples)
             train_dataset = train_dataset.select(range(num_calibration_samples))
             quantization_config.calibration_sampling_size = num_calibration_samples
+        elif optim_args.quantization_approach == "weight_only":
+            train_dataset = train_dataset.select(range(num_calibration_samples))
 
         quantizer.quantize(
             quantization_config=quantization_config,
             save_directory=training_args.output_dir,
             calibration_dataset=train_dataset
             if optim_args.quantization_approach in ["static", "weight_only"]
             else None,
-            batch_size=1  # batch_size > 1 for GPTQ is WIP
-            if optim_args.quantization_approach == "weight_only" and optim_args.quantization_methodology == "GPTQ"
+            batch_size=1
+            if optim_args.quantization_approach == "weight_only"
             else training_args.per_device_train_batch_size,
-            weight_only=True if optim_args.quantization_approach == "weight_only" else False,
         )
         trainer.model = quantizer._quantized_model
-    if optim_args.apply_quantization and optim_args.verify_loading:
+
+    # Weight only quantization didn't support save/load function due to weight only model has private linear operator.

```

Comment:
This is an issue no ? If wa can't save / load the model, what about the calibration step ? (not something that we want to perform each time we need to laod the model)

Reply:
Yes, "save/load" is a TODO job, In the initial design, itrex weight-only quantization only supports the RTN algorithm, which does not require calibration.Later, algorithms such as AWQ and TEQ were added, and these two algorithms require calibration.Therefore, we will add save/load methods in the future.